pandas/tests/dtypes/test_inference.py

```diff
@@ -1582,6 +1582,31 @@ def test_is_string_array(self):
         )
         assert not lib.is_string_array(np.array([1, 2]))
 
+    @pytest.mark.parametrize(
+        "func",
+        [
+            "is_bool_array",
+            "is_date_array",
+            "is_datetime_array",
+            "is_datetime64_array",
+            "is_float_array",
+            "is_integer_array",
+            "is_interval_array",
+            "is_string_array",
+            "is_time_array",
+            "is_timedelta_or_timedelta64_array",
+        ],
+    )
+    def test_is_dtype_array_empty_obj(self, func):
```

Comment:
Do we have test coverage for `is_object_array` returning `True` explicitly elsewhere?

Reply:
I'm not seeing an `is_object_array`.

Reply:
Ah my mistake - I think I got tripped up the PR title. That should say `is_*_array returns false on empty object dtype` instead of `true` right? Or am I overlooking something?

Reply:
I suppose it depends how you want the title of a PR constructed. The bug is that it returns true, this PR's title is a statement of the bug.